SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift

```diff
@@ -0,0 +1,161 @@
+//===--- LifetimeDependenceScopeFixup.swift ----------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SIL
+
+
+private let verbose = false
+
+private func log(_ message: @autoclosure () -> String) {
+  if verbose {
+    print("### \(message())")
+  }
+}
+
+let lifetimeDependenceScopeFixupPass = FunctionPass(
+  name: "lifetime-dependence-scope-fixup")
+{ (function: Function, context: FunctionPassContext) in
+  log("Scope fixup for lifetime dependence in \(function.name)")
+
+  for instruction in function.instructions {
+    guard let markDep = instruction as? MarkDependenceInst else { continue }
```

Comment:
formatting: it is more readable to put the `continue` in a separate line. Like swift-format would do.